### PR TITLE
Document the requirements for onboarding a component

### PR DESCRIPTION
**Pull request by peternied**
Previously we did not have explicit documentation of the requirements for components in the distribution build.  Creating a starting place for these requirements for the 1.1.0 release.

### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: bundle-workflow/README.md
Comment (mch2) [inline bundle-workflow/README.md:223]: nit `streamlined`

File: bundle-workflow/README.md
Comment (mch2) [inline bundle-workflow/README.md:225]: `appropriately`

File: bundle-workflow/README.md
Comment (mch2) [inline bundle-workflow/README.md:226]: `respond`

File: bundle-workflow/README.md
Comment (mch2) [inline bundle-workflow/README.md:230]: This validation should `be performed`

**Review by mch2 — CHANGES_REQUESTED** [review]

File: bundle-workflow/README.md
Comment (dblock) [inline bundle-workflow/README.md:225]: issue -> issues

File: bundle-workflow/README.md
Comment (dblock) [inline bundle-workflow/README.md:226]: I would say "Component maintainers regularly triage their issues and respond to issues blocking the bundle workflow promptly.". If you're going to impose an SLA we need to discuss this with the said component owners first.

File: bundle-workflow/README.md
Comment (dblock) [inline bundle-workflow/README.md:225]: I would prefer not to use the word "must", because everything is a must here. Just say "Component reponsitories create tags that match ..., for example ..."

File: bundle-workflow/README.md
Comment (dblock) [inline bundle-workflow/README.md:227]: This should say "in their own repos" or something clearer.

File: bundle-workflow/README.md
Comment (dblock) [inline bundle-workflow/README.md:228]: This is not a requirement, it's a statement of what happens when things break, remove.

File: bundle-workflow/README.md
Comment (dblock) [inline bundle-workflow/README.md:233]: It's a non-requirement :)

**Review by dblock — CHANGES_REQUESTED** [review]
I think I'd prefer something a bit shorter, sweeter and clearer, with examples. For example ...

```
- Passing CI in GitHub Actions that runs unit and integration tests. See job-scheduler CI workflow (link) for an example.
-    
```

**Review by gaiksaya — APPROVED** [review]

File: bundle-workflow/README.md
Comment (peternied) [reply bundle-workflow/README.md:226]: Nixing this section out - feels heavy handed on re-read

File: bundle-workflow/README.md
Comment (peternied) [reply bundle-workflow/README.md:225]: labelled! Thanks

**Review by dblock — DISMISSED** [review]

File: bundle-workflow/README.md
Comment (peternied) [inline bundle-workflow/README.md:262]: Going to add a template for this request

File: bundle-workflow/README.md
Comment (peternied) [inline bundle-workflow/README.md:253]: This is complex to validate and possibility do it wrong, maybe we could add a snapshot artifact validation script and update the step

**Review by peternied — COMMENTED** [review]
Merged in @mch2 changes, lets keep working in.  Adopted some of the recommendations and added some of my own next steps while we workshop the onboarding process

**Comment by peternied** [comment]
@dblock @gaiksaya Take another look?

**Comment by dblock** [comment]
Needs a rebase.

File: bundle-workflow/README.md
Comment (dblock) [inline bundle-workflow/README.md:247]: Lowercase add. I would actually change this to “ensure” because we use plug-in meta to create labels across all our component repos. You can link that but it’s probably overkill. Either way use a component repo as an example and link that label example to issues labeled as such in  that repo.

**Review by gaiksaya — APPROVED** [review]

File: bundle-workflow/README.md
Comment (dblock) [inline bundle-workflow/README.md:251]: The location has changed to different defaults for OpenSearch and OpenSearch Dashboards.

File: bundle-workflow/README.md
Comment (dblock) [reply bundle-workflow/README.md:253]: This only matters for OpenSearch. Sounds like we may want two subsections.

**Review by dblock — CHANGES_REQUESTED** [review]

File: bundle-workflow/README.md
Comment (dblock) [inline bundle-workflow/README.md:321]: This location is still wrong to build.sh

File: bundle-workflow/README.md
Comment (dblock) [inline bundle-workflow/README.md:315]: Add to TOC

**Review by dblock — CHANGES_REQUESTED** [review]

**Comment by abhinavGupta16** [comment]
@peternied - Do we have an issue/story to which this PR tends to? If yes, can we that to this PR?

File: bundle-workflow/README.md
Comment (dblock) [inline bundle-workflow/README.md:329]: Sounds like you need two sections :)

**Review by dblock — DISMISSED** [review]

File: bundle-workflow/README.md
Comment (peternied) [reply bundle-workflow/README.md:329]: I think we will need to break these apart more when we have a OpenSearch Dashboards workflow, ideally we can automate the validation of this (have a way to check if the artifact has an incorrect name).

File: bundle-workflow/README.md
Comment (saratvemulapalli) [inline bundle-workflow/README.md:316]: Looks like we are duplicating information. 
Most of this is already captured in https://github.com/opensearch-project/opensearch-build/blob/main/ONBOARDING.md. 

If it is for component owners, I would rather let it go in ONBOARDING.md as a single place to look for and this README.md can link to it if needed.

**Review by dblock — APPROVED** [review]

**Comment by peternied** [comment]
Lets defer to @saratvemulapalli onboarding guide in #559, we can circle back if we need more details.

**Pull request closed without merging**
